src/pyvesync/vesync.py

```diff
@@ -221,15 +234,18 @@ def update(self):
                     self.last_update_ts = time.time()
 
     def check_response(self, resp, call):
-        stand_resp = ['get_devices', 'login', '15a_detail', '15a_toggle',
+        stand_resp = ['get_devices', '15a_detail', '15a_toggle',
                       '15a_energy', 'walls_detail', 'walls_toggle',
                       '10a_detail', '10a_toggle', '10a_energy', '15a_ntlight'
                       ]
-        if call in stand_resp:
-            if resp['code'] == 0:
+        if call in stand_resp and isinstance(resp, dict):
+            if 'code' in resp and resp['code'] == 0:
                 return True
             else:
                 return False
+        elif call == 'login':
+            if resp['code'] == 0 and 'result' in resp:
```

Comment:
You should probably factor out the `isinstance(resp, dict)` check, as this branch could fail as well.

Reply:
Nice catch, made the changes